clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -957,6 +958,9 @@ public void assign(Collection<TopicPartition> partitions) {
             if (timeout < 0)
                 throw new IllegalArgumentException("Timeout must not be negative");
 
+            if (this.subscriptions.hasNoSubscriptionOrUserAssignment())
+                throw new IllegalPollException("Consumer is not subscribed to any topics or assigned any partitions");
```

Comment:
Sure, I'll modify it if we decide that we want to go ahead with this solution. Thanks.